packages/code-studio/src/styleguide/StyleGuide.tsx

```diff
@@ -76,63 +79,91 @@ function StyleGuide(): React.ReactElement {
           <h1 style={{ paddingTop: '2rem' }}>Deephaven UI Components</h1>
         </Flex>
 
-        <Flex
-          {...stickyProps}
-          UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
-          marginTop={-56}
-          top={20}
-          gap={10}
-          alignItems="end"
-        >
-          {hasMultipleThemes ? <ThemePicker /> : null}
-          <SamplesMenu />
-        </Flex>
-        <Flex
-          {...stickyProps}
-          UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
-          top="calc(100vh - 40px)"
-          marginTop={-32}
-          marginEnd={hasMultipleThemes ? -234 : 0}
-        >
-          <GotoTopButton />
-        </Flex>
+        {isHash('') && (
+          <>
+            <Flex
+              {...stickyProps}
+              UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
+              marginTop={-56}
+              top={20}
+              gap={10}
+              alignItems="end"
+            >
+              {hasMultipleThemes ? <ThemePicker /> : null}
+              <SamplesMenu />
+            </Flex>
+            <Flex
+              {...stickyProps}
+              UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
+              top="calc(100vh - 40px)"
+              marginTop={-32}
+              marginEnd={hasMultipleThemes ? -234 : 0}
+            >
+              <GotoTopButton />
+            </Flex>
+          </>
+        )}
 
-        <Typograpy />
+        {isHash('typography') && <Typograpy />}
 
-        <SampleMenuCategory data-menu-category="Colors" />
-        <Colors />
-        <ThemeColors />
+        {isHash('') && <SampleMenuCategory data-menu-category="Colors" />}
+        {isHash('colors') && <Colors />}
+        {(isHash('theme-color-palette') ||
```

Comment:
I would consider allowing multiple values passed to `isHash` instead of || chaining